### PR TITLE
Bidirectional lstm example #2093

### PR DESCRIPTION
**Pull request by xlvector**
This is an just simple example for bidirectional lstm.

### CODE REVIEW

**Comment by piiswrong** [comment]
@pluskid 
We need a LSTM symbol implemented in python under mx.sym.LSTM so that it will be easier to replace LSTM in all examples with cudnn v5 lstm when we support it

**Comment by pluskid** [comment]
@piiswrong yes agreed. The currently used LSTM in ptb and speech are a variant with some simplification. An implementation of a full version needs the broadcast multiplication feature. Need to check what variant of the LSTM cell is cuDNN v5 providing. What is our current status of supporting cuDNN v5? I am reluctant to upgrade having heard that they breaks a lot of APIs....

**Comment by antinucleon** [comment]
@pluskid We now support everything but LSTM cell in CuDNN v5. CUDA 7.5 is fine for CuDNN v5.

File: make/config.mk [hunk 1]
```diff
@@ -28,33 +28,33 @@ export NVCC = nvcc
 DEBUG = 0
 
 # the additional link flags you want to add
-ADD_LDFLAGS =
+ADD_LDFLAGS = -L/disk1/deeplearning/local_install/lib
```
Comment (antinucleon) [inline make/config.mk]: This file should not be modified.

File: make/config.mk [hunk 2]
```diff
@@ -28,33 +28,33 @@ export NVCC = nvcc
...
 
 # the additional compile flags you want to add
-ADD_CFLAGS =
+ADD_CFLAGS = -I/disk1/deeplearning/local_install/include
```
Comment (antinucleon) [inline make/config.mk]: same

File: make/config.mk [hunk 3]
```diff
@@ -28,33 +28,33 @@ export NVCC = nvcc
...
 
 #---------------------------------------------
 # matrix computation libraries for CPU/GPU
 #---------------------------------------------
 
 # whether use CUDA during compile
-USE_CUDA = 0
+USE_CUDA = 1
```
Comment (antinucleon) [inline make/config.mk]: same

File: make/config.mk (hunk 1 above)
Comment (xlvector) [inline make/config.mk]: thanks, I will modify it

**Comment by xlvector** [comment]
Glad to hear that mxnet will have a full support of LSTM.

Another think I want to ask is, is it easy to integrate baidu wrap-ctc without using torch module?

**Comment by tqchen** [comment]
this is ready to be merged after combine the commits. see instructions http://mxnet.readthedocs.io/en/latest/how_to/contribute.html#how-to-combine-multiple-commits-into-one

**Comment by xlvector** [comment]
sorry to response later. I combined the commits but find its already merged. 

Thanks all.

**Comment by jostep** [comment]
@xlvector I am working on make this example python2/3 compatible. However, I find that the data folder and gen_data.py are missing in the current master branch. Could you take a look at this?

**Comment by xlvector** [comment]
Thanks. The gen_data.py is very simple, it just generate whitespace seprated numbers in every line.  

something like

for i in range(1000):
    print ' '.join([x for x in sorted([random.randint(0, 100) for k in range 10)]

You can add this script in your PR.

**Comment by jostep** [comment]
ok I will put this gen_data into the lstm_sort.py directly. Thanks.

**Pull request merged**